<?xml version="1.0" encoding="utf-8"?>
<launch>
    <!-- set all the arguments default values -->
    <arg name="object_file" default="shape11_high_res.pcd"/>
    <arg name="file_path" default="$(find shape_analysis)/shapes/"/>
    <arg name="fingers_length" default="0.04"/>
    <arg name="voxel_res" default="1.0"/>
    <arg name="seed_res" default="10.0"/>
    <arg name="color_imp" default="0.0"/>
    <arg name="spatial_imp" default="1.0"/>
    <arg name="normal_imp" default="1.0"/>
    <arg name="supervoxel_iterations" default="10"/>
    <arg name="disable_transf" default="true"/>
    <arg name="dmg_service_name" default="in_hand_path"/>

    <!-- start the helper servers-->
    <include file="$(find shape_analysis)/launch/help_servers.launch"/>

    <!-- run the generation of the dmg-->
    <node pkg="shape_analysis" type="extendedDMG_server" name="extendedDMG_server" output="screen">
        <param name="object_name" value="$(arg object_file)"/>
        <param name="object_file_path" value="$(arg file_path)"/>
        <param name="finger_length" value="$(arg fingers_length)"/>
        <param name="voxel_resolution" value="$(arg voxel_res)"/>
        <param name="seed_resolution" value="$(arg seed_res)"/>
        <param name="color_importance" value="$(arg color_imp)"/>
        <param name="spatial_importance" value="$(arg spatial_imp)"/>
        <param name="normal_importance" value="$(arg normal_imp)"/>
        <param name="refinement_iterations" value="$(arg supervoxel_iterations)"/>
        <param name="disable_transform" value="$(arg disable_transf)"/>
        <param name="service_name" value="$(arg dmg_service_name)"/>
    </node>
</launch>